src/context/instanceLoading.tsx

```diff
@@ -1,53 +1,44 @@
-import React, {
-  createContext,
-  FC,
-  ReactNode,
-  useContext,
-  useState,
-} from "react";
-import { LxdInstance } from "types/instance";
+import React, { createContext, FC, ReactNode, useContext } from "react";
 
-type LoadingTypes = "Starting" | "Stopping" | "Restarting" | "Pausing";
+type LoadingType =
+  | "Creating"
+  | "Starting"
+  | "Stopping"
+  | "Restarting"
+  | "Pausing";
 
 export interface InstanceLoadingType {
-  getType: (instance: LxdInstance) => LoadingTypes | undefined;
-  setLoading: (instance: LxdInstance, loadingType: LoadingTypes) => void;
-  setFinish: (instance: LxdInstance) => void;
+  get: (instanceName: string) => LoadingEntry | undefined;
+  set: (name: string, type: LoadingType, callback?: () => void) => void;
+  finish: (instanceName: string) => void;
 }
 
 const InstanceLoadingContext = createContext<InstanceLoadingType>({
-  getType: () => undefined,
-  setLoading: () => undefined,
-  setFinish: () => undefined,
+  get: () => undefined,
+  set: () => undefined,
+  finish: () => undefined,
 });
 
 interface Props {
   children: ReactNode;
 }
 
-export const InstanceLoadingProvider: FC<Props> = ({ children }) => {
-  const [instanceStates, setInstanceStates] = useState(
-    new Map<string, LoadingTypes>()
-  );
-
-  const setLoading = (instance: LxdInstance, loadingType: LoadingTypes) => {
-    const newMap = new Map(instanceStates);
-    newMap.set(instance.name, loadingType);
-    setInstanceStates(newMap);
-  };
+interface LoadingEntry {
+  type: LoadingType;
+  callback?: () => void;
+}
 
-  const setFinish = (instance: LxdInstance) => {
-    const newMap = new Map(instanceStates);
-    newMap.delete(instance.name);
-    setInstanceStates(newMap);
-  };
+const sharedMap = new Map<string, LoadingEntry>();
```

Comment:
This needs to go in the state just like it used to. Currently the loading logic is broken - you'll always get `false` for `isLoading`, and the spinner never appears:

![Peek 2023-05-29 21-27](https://github.com/canonical/lxd-ui/assets/56583786/720c252b-146e-468e-8a50-2ffc78fce15c)